### PR TITLE
Further wordsmithing of the postinit section

### PR DESCRIPTION
**Pull request by bradcray**
Tried to rephrase this text to make postinit in 1.17 seem less like something that's barely working and more like something that adds value and seems to be working as we'd expect.  Other minor local tweaks while here.

### CODE REVIEW

**Comment by bradcray** [comment]
@benharsh, @lydia-duncan, @noakesmichael : Can you let me know what you think of this?  (I was concerned that the text remained too apologetic and version-specific for something that seems to largely be working as we'd like).

File: doc/rst/technotes/initializers.rst [hunk 1]
```diff
@@ -474,22 +474,15 @@ Post Initialization for Records
...
-compiler defaults to initializers rather than constructors, the
-``postinit`` method will provide a convenient way to perform final
-updates to an instance of a record when the default all-fields
-initializer is sufficient.
+A user may override this method and customize the behavior.  Writing a
+``postinit`` method provides a way for the record author to leverage
+the default all-fields initializer while also specifying additional
+computation to perform before returning the new object.  When used
+with class hierarchies, the ``postinit`` method offers additional
+benefits (see :ref:`post-init-classes`) below).
```
Comment (lydia-duncan) [inline doc/rst/technotes/initializers.rst]: Stray closing parenthesis here.  Seems fine to me otherwise, though!

**Comment by bradcray** [comment]
Removed a forward reference that broke a principle that Mike was trying to follow of keeping class-specific content (or forward references to same) out of the records section.  Also happens to fix the too-many-parens error Lydia caught.

**Comment by benharsh** [comment]
Looks good to me!

**Pull request merged**